_appmap/instrument.py

```diff
@@ -101,7 +101,7 @@ def call_instrumented(f, instance, args, kwargs):
         return ret
     except AppMapLimitExceeded:
         raise
-    except Exception:  # noqa: E722
+    except BaseException:  # noqa: E722
```

Comment:
Looking at the standard library's exception hierarchy, it's not clear to me why this change fixes the problem. Is there some other exception that inherits from `BaseException` that's getting raised?

Reply:
In `pytest` project a `Skipped` exception is raised in `skip` function here: https://github.com/pytest-dev/pytest/blob/main/src/_pytest/outcomes.py#L160. However, `Skipped` does not derive from `Exception`, it is (indirectly) derived from `BaseException`. We were not catching it in `_appmap/instrument.py::call_instrumented` and we were not able to create the return leg of the call, the exception return event. Missing this event resulted in a "bad" AppMap.

Reply:
Thanks for this explanation. Can you add a skipped test to https://github.com/getappmap/appmap-python/blob/master/_appmap/test/data/pytest/test_simple.py, please, to show that you've fixed the problem that was originally reported? Thanks!